src/ExtUvLoop.php

```diff
@@ -313,4 +314,26 @@ private function createStreamListener()
 
         return $callback;
     }
+
+    /**
+     * @param float $interval
+     * @return int
+     */
+    private function convertFloatSecondsToMilliseconds($interval)
+    {
+        if ($interval < 0) {
+            return 0;
+        }
+
+        $maxValue = (int) (\PHP_INT_MAX / 1000);
+        $intInterval = (int) $interval;
+
+        if ($intInterval >= $maxValue) {
```

Comment:
I've just tested this and unfortunately there's an edge case, which makes the comparison fail, even though the interval is significantly larger (as float) than the PHP integer max value.

Based on <https://3v4l.org/TIkG3>, the suggestion should work around that. You should probably add test cases for these.

```suggestion
        if ($interval > \PHP_INT_MAX || ($intInterval <= 0 && $interval > 0)) {
```

Reply:
You are right, fixed and tested.